MyStore/src/MyStore.cpp

```diff
@@ -3,35 +3,61 @@
 #include <chrono>
 #include <windows.h>
 #include <filesystem>
+#include <thread>
+#include <mutex>
 
 using namespace std::chrono;
 
 // This variable is needed to avoid confusion with relative paths in different modules
 // Example of usage:
 // std::string filePath = gProjectPath + "\\data\\Data.json"
+
+void GenerateProducts(StorageData& stor, const int& period, bool& flag);
+void ExitCondition(bool& flag);
 const std::string gProjectPath = std::filesystem::current_path().generic_string();
+std::mutex mtx;
 
 int main(int argc, char* argv[])
 {
 	StorageData storage;
+	bool flag_thread = false;
 	const int period_for_element_milliseconds = Settings::Instance().GetTimeGenerate();
 
 	storage.m_Parcer.Deserialization();
 	storage.m_Parcer.ShowData();
 
+	std::thread T1(GenerateProducts, std::ref(storage), std::ref(period_for_element_milliseconds), std::ref(flag_thread));
+	std::thread T2(ExitCondition, std::ref(flag_thread));
+	T1.join();
+	T2.join();
+
+	return 0;
+}
+
+void GenerateProducts(StorageData& stor, const int& period, bool& flag)
+{
+	mtx.lock();

```

Comment:
Here we don't need mutex since the second thread doesn't access the same data and we can't run into race conditions, so let's remove it. But it's good that you thought about it. :)